src/App.tsx

```diff
@@ -15,47 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reverse = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.currentTarget.classList.toggle('is-light');
+    setIsReversed(!isReversed);
+  };
+
+  const visibleGoods = [...goodsFromServer];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 1;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={
+            classNames('button is-info',
+              {
+                'is-light': sortType !== SortType.ALPHABET,
+              })
+          }
+          onClick={() => (setSortType(1))}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={
+            classNames('button is-success',
+              {
+                'is-light': sortType !== SortType.LENGTH,
+              })
+          }
+          onClick={() => (setSortType(2))}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={
+            classNames('button is-warning',
+              {
+                'is-light': !isReversed,
+              })
+          }
+          onClick={reverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        { ((sortType !== SortType.NONE || isReversed)) && (
```

Comment:
```suggestion
        {(sortType !== SortType.NONE || isReversed) && (
```
1 brackets pair is redundant